Entity/CategoryManager.php

```diff
@@ -136,6 +176,31 @@ public function getRootCategories($loadChildren = true)
     }
 
     /**
+     * @param bool|true $loadChildren
+     *
+     * @return array
+     */
+    public function getRootCategoriesSplitByContexts($loadChildren = true)
```

Comment:
`getRootCategory` and `getRootCategories ` both return only one root category for the context.
For supporting several root categories I added:

- `getAllRootCategories` - replacement for `getRootCategories`
- `getRootCategoriesSplitByContexts` - for `CategoryAdminController`
- `getRootCategoriesForContext` - replacement for `getRootCategory` in `CategorySelectorType`
- `getRootCategoryWithChildren` - for loading children categories if they are not loaded yet. Also replacement of `getRootCategory`